pkg/migration/controllers/migration.go

```diff
@@ -278,13 +416,83 @@ func (m *MigrationController) migrateVolumes(migration *stork_api.Migration) err
 		}
 	}
 
-	err = sdk.Update(migration)
+	err := sdk.Update(migration)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+func (m *MigrationController) runPreExecRule(migration *stork_api.Migration) ([]chan bool, error) {
+	if migration.Spec.PreExecRule == "" {
+		migration.Status.Stage = stork_api.MigrationStageVolumes
+		migration.Status.Status = stork_api.MigrationStatusPending
+		err := sdk.Update(migration)
+		if err != nil {
+			return nil, err
+		}
+		return nil, nil
+	} else if migration.Status.Stage == stork_api.MigrationStageInitial {
+		migration.Status.Stage = stork_api.MigrationStagePreExecRule
+		migration.Status.Status = stork_api.MigrationStatusPending
+	}
+
+	if migration.Status.Stage == stork_api.MigrationStagePreExecRule {
+		if migration.Status.Status == stork_api.MigrationStatusPending {
+			if migration.Spec.PreExecRule != "" {
```

Comment:
No need for this check as we won't reach here if pre exec rule is empty

Reply:
Removed